packages/mco/components/data-policies/drpolicy-list-page/application-status.tsx

```diff
@@ -0,0 +1,255 @@
+import * as React from 'react';
+import { useK8sGet } from '@odf/shared/hooks/k8s-get-hook';
+import { ModalBody, ModalFooter } from '@odf/shared/modals/Modal';
+import { ApplicationModel } from '@odf/shared/models/common';
+import { ListKind } from '@odf/shared/types';
+import { ApplicationKind } from '@odf/shared/types/k8s';
+import { useTranslation } from 'react-i18next';
+import { 
+    Modal,
+    Button,
+    ButtonVariant,
+    ModalVariant,
+    Text,
+    TextContent,
+    TextVariants,
+    Toolbar,
+    ToolbarContent,
+    ToolbarItem,
+    SearchInput,
+    Bullseye,
+} from '@patternfly/react-core';
+import { TableComposable, Tbody, Tr, Td, Th, Thead, ThProps} from '@patternfly/react-table';
+import { ACMPlacementRuleModel, ACMSubscriptionModel } from '../../../models/models';
+import { ACMSubscriptionKind, DRPlacementControlKind} from '../../../types/types';
+import { matchApplicationsToSubstring } from '../../../utils';
+import './application-status.scss';
+
+
+const reactPropFix = {
+    translate: 'yes',
+};
+
+type ApplicationStatusPros = {
+    drPlacementControls: DRPlacementControlKind[];
+};
+
+type ApplicationMapping = {
+    [namespace in string]: {
+        [app in string] : ApplicationKind;
+    };
+};
+
+type SubcsriptionMapping = {
+    [namespace in string]: {
+        [sub in string] : ACMSubscriptionKind
+    };
+};
+
+type PlacementRuleNameMapping = {
+    [namespace in string]: string[];
+};
+
+const appFilter = (application: ApplicationKind) => (
+    application?.spec?.componentKinds.some((componentKind) => 
```

Comment:
```suggestion
    application?.spec?.componentKinds?.some((componentKind) => 
```
most probably it will work without adding it as well, but just to be sure in case ` application?.spec?.componentKinds` is `undefined`.